packages/react/src/components/List/List.types.ts

```diff
@@ -139,6 +139,21 @@ export interface IListProps<T = any> extends React.HTMLAttributes<List<T> | HTML
    */
   onRenderCell?: (item?: T, index?: number, isScrolling?: boolean) => React.ReactNode;
 
+  /**
+   * Method to call when trying to render an item conditionally.
+   *
+   * When this method returns `null` the cell will be skipped in the render.
+   *
+   * This prop is mutually exclusive with `onRenderCell` and when `onRenderCellConditional` is set,
+   * `onRenderCell` will not be called.
+   *
+   * @param item - The data associated with the cell that is being rendered.
+   * @param index - The index of the cell being rendered.
+   * @param isScrolling - True if the list is being scrolled. May be useful for rendering a placeholder if your cells
+   * are complex.
+   */
+  onRenderCellConditional?: (item?: T, index?: number, isScrolling?: boolean) => React.ReactNode | null;
```

Comment:
Since you are adding a capability to List, should and example of using this be added to List's documentation?

Reply:
Yes. Since GroupedListV2 is still preview I'll add these docs in a follow up: #24819